src/o365/graph/commands/reports/reports-teams-device-usage-user-detail.ts

```diff
@@ -0,0 +1,153 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import request from '../../../../request';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+const maxDays = 28;
+const maxDate: Date = new Date();
+maxDate.setDate(maxDate.getDate() - maxDays);
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  period?: string;
+  date?: string;
+}
+
+class GraphReportsTeamsDeviceUsageUserDetailCommand extends GraphItemsListCommand<any> {
```

Comment:
Since we're not retrieving multiple items but calling a single report, we should inherit from the `GraphCommand` base class which is better suited for this purpose.

Reply:
Fixed and ready for review.